atom/browser/atom_download_manager_delegate.cc

```diff
@@ -28,6 +32,34 @@ namespace {
 const DownloadPathReservationTracker::FilenameConflictAction
     kDefaultPlatformConflictAction = DownloadPathReservationTracker::UNIQUIFY;
 
+NativeWindow* GetLastActiveWindow() {
+  BrowserList* browser_list = BrowserList::GetInstance();
+  // Get recently active browser.
+  Browser* browser = browser_list->GetLastActive();
+  // During the startup this might be null but I'm not sure this can be called
+  // at that time.
+  if (!browser)
+    browser = *browser_list->begin();
+  return static_cast<NativeWindow*>(browser->window());
+}
+
+NativeWindow* GetNativeWindowFromWebContents(content::WebContents* web_contents) {
+  DCHECK(web_contents);
+  auto relay = NativeWindowRelay::FromWebContents(web_contents);
+  return relay ? relay->window.get() : nullptr;
+}
+
+// Blank newtab on download should be closed after starting download.
+void CloseWebContentsIfNeeded(content::WebContents* web_contents) {
+  DCHECK(web_contents);
+  Browser* browser = chrome::FindBrowserWithWebContents(web_contents);
+  if (browser && web_contents->GetController().IsInitialNavigation() &&
```

Comment:
are these extra checks necessary? I don't think we ever want to leave a single blank tab from a download

Reply:
Just in case download is fired on browser startup, we shouldn't close window. This might happen when browser is closed during the download accidentally.